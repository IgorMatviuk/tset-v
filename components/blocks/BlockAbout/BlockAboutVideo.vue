<template>
  <div class="about-video">
    <ImageLoader>
      <template v-if="content.video">
        <client-only>
          <LazyVideo
            :src="content.video"
            :poster="content.videoPoster"
            :attrs="attrs"
            :pause-on-exit="!$device.isMobile"
          />
        </client-only>
      </template>
      <template v-else>
        <img
          :data-src="content.videoPoster"
          data-object-fit="cover"
          data-object-position="center center"
          :alt="content.title"
          class="about-video__item lazyload"
          width="100"
          height="100"
          loading="lazy"
        />
      </template>
    </ImageLoader>
  </div>
</template>

<script>
import ImageLoader from '@/components/shared/ImageLoader'

export default {
  components: {
    ImageLoader
  },
  props: {
    content: {
      type: Object,
      required: true
    }
  },
  data() {
    return {
      attrs: {
        preload: 'none',
        muted: true,
        loop: true,
        autoplay: !this.$device.isMobile,
        playsinline: this.$device.isMobile,
        controls: this.$device.isMobile,
        class: 'about-video__item',
        'data-object-fit': 'cover',
        'data-object-position': 'center'
      }
    }
  }
}
</script>

<style lang="scss">
.about-video {
  position: relative;
  width: 100%;
  height: 0;
  padding-bottom: 68.1159%;

  &__item {
    position: absolute;
    top: 0;
    left: 0;
    object-fit: cover;
    object-position: center;
    width: 100%;
    min-width: 100%;
    height: 100%;
  }

  @include media-breakpoint-up(md) {
    padding-bottom: 66.9%;
  }

  @include media-breakpoint-up(xxl) {
    padding-bottom: 66.7%;
  }
}
</style>
