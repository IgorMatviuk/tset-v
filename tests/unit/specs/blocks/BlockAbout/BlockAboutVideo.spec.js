import { mount, createLocalVue } from '@vue/test-utils'
import VueLazyLoadVideo from 'vue-lazyload-video'

import BlockAboutVideo from '@/components/blocks/BlockAbout/BlockAboutVideo'

const localVue = createLocalVue()
localVue.use(VueLazyLoadVideo)

describe('BlockAboutVideo', () => {
  test('has expected structure: without video', () => {
    const wrapper = mount(BlockAboutVideo, {
      propsData: {
        content: {
          title: 'some title',
          videoPoster: '/some-url'
        }
      },
      mocks: {
        $device: {
          isMobile: true
        }
      },
      stubs: ['client-only', 'LazyVideo']
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('has expected structure: with video', () => {
    const wrapper = mount(BlockAboutVideo, {
      propsData: {
        content: {
          title: 'some title',
          video: '/some-url',
          videoPoster: '/some-url'
        }
      },
      mocks: {
        $device: {
          isMobile: false
        }
      },
      stubs: ['client-only', 'LazyVideo']
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('render video', () => {
    const wrapper = mount(BlockAboutVideo, {
      localVue,
      propsData: {
        content: {
          title: 'some title',
          video: '/some-url',
          videoPoster: '/some-url'
        }
      },
      mocks: {
        $device: {
          isMobile: true
        }
      },
      stubs: ['client-only']
    })

    expect(wrapper.find('video').exists()).toBe(true)
  })
})
