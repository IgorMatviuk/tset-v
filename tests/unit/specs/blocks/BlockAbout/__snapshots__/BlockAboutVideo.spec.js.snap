// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockAboutVideo has expected structure: with video 1`] = `
<div
  class="about-video"
>
  <div
    class="img-loader"
  >
    <client-only-stub>
      <lazyvideo-stub
        attrs="[object Object]"
        pause-on-exit="true"
        poster="/some-url"
        src="/some-url"
      />
    </client-only-stub>
  </div>
</div>
`;

exports[`BlockAboutVideo has expected structure: without video 1`] = `
<div
  class="about-video"
>
  <div
    class="img-loader"
  >
    <img
      alt="some title"
      class="about-video__item lazyload"
      data-object-fit="cover"
      data-object-position="center center"
      data-src="/some-url"
      height="100"
      loading="lazy"
      width="100"
    />
  </div>
</div>
`;
